src/Implement/UnitTestHttpListener.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <arpa/inet.h>
+#include <unistd.h>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "HttpListener.h"
+#include <iostream>
+class TestCallback{
+
+  public:
+    static void TestCallbackFunction(const int &fd){
+      TestCallback::test_callback = true;
+    }
+    static bool test_callback;
+};
+
+bool TestCallback::test_callback = false;
+
+void TestClient(){
+  int sockfd = 0;
+  sockfd = socket(AF_INET , SOCK_STREAM , 0);
+  struct sockaddr_in info;
+  bzero(&info,sizeof(info));
+  info.sin_family = PF_INET;
+  info.sin_addr.s_addr = inet_addr("127.0.0.1");
+  info.sin_port = htons(9000);
```

Comment:
Consider pulling this out as a global anonymous constant so that you will not change one and not the other.
```
namespace {
// Test TCP Port for testing the listener.
constexpr uint16_t kTestPort = 9000;
}
```

Reply:
Done